riocli/deployment/execute.py

```diff
@@ -22,44 +22,77 @@
 else:
     from riocli.utils.spinner import DummySpinner as Spinner
 
-from riocli.constants import Colors
-from riocli.deployment.util import select_details
-from riocli.utils.execute import run_on_cloud
-
+from riocli.config import new_v2_client
+from riocli.config import new_client
+from riocli.constants import Colors, Status
+from riocli.utils.execute import run_on_device
+from riocli.device.util import find_device_by_name
 
 @click.command(
     'execute',
     cls=HelpColorsCommand,
     help_headers_color=Colors.YELLOW,
     help_options_color=Colors.GREEN,
 )
-@click.option('--component', 'component_name', default=None,
-              help='Name of the component in the deployment')
+@click.option('--user', default='root')
+@click.option('--shell', default='/bin/bash')
 @click.option('--exec', 'exec_name', default=None,
               help='Name of a executable in the component')
 @click.argument('deployment-name', type=str)
 @click.argument('command', nargs=-1)
-# @name_to_guid
 def execute_command(
-        component_name: str,
+        user: str,
+        shell: str,
         exec_name: str,
         deployment_name: str,
-        deployment_guid: str,
         command: typing.List[str]
 ) -> None:
     """
-    Execute commands on cloud deployment
+    Execute commands on device deployment
     """
     try:
-        comp_id, exec_id, pod_name = select_details(deployment_guid, component_name, exec_name)
+        client = new_v2_client()
+
+        deployment = client.get_deployment(deployment_name)
+        if not deployment:
+            click.secho(f'Deployment `{deployment_name}` not found', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.status.status != Status.RUNNING:
+            click.secho(f'Deployment `{deployment_name}` is not in Running state', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        if deployment.spec.runtime != 'device':
+            click.secho(f'Execution on cloud deployments is not supported', fg=Colors.RED)
+            raise SystemExit(1)
+        
+        package = client.get_package(deployment.metadata.depends.nameOrGUID, query={"version": deployment.metadata.depends.version})
```

Comment:
Move this line inside `exec_name is None` check. You don't want to fetch if not needed.

Reply:
now we will need it outside to verify if the user package contains exec_name provided by user.
In case user hasn't provided exec_name it will be needed to populate the exec_name list.